runtime/gc_vlhgc/SchedulingDelegate.cpp

```diff
@@ -524,44 +663,195 @@ MM_SchedulingDelegate::calculateEstimatedGlobalBytesToScan() const
 	return liveSetAdjustedForScannableBytesRatio;
 }
 
-UDATA
+intptr_t
+MM_SchedulingDelegate::calculateRecommendedEdenChangeForExpandedHeap(MM_EnvironmentVLHGC *env) 
+{
+
+	if (0 == _pgcCountSinceGMPEnd) {
+		/* No statistics have been collected - just return the current eden size */
+		return getCurrentEdenSizeInBytes(env);
+	}
+	
+	/* 
+	 * Several statistics have observed which are needed to predict best eden size. 
+	 * These statistics are used to predict what eden size will lead to the lowest overhead, where overhead, is a hybrid 
+	 * between % of time spent in gc, and pgc pause times. The goal is to minimize % of time spent in gc,
+	 * while staying below the specific gc pause time threshold
+	 */
+
+	uint64_t avgPgcTimeUs = _historicalPartialGCTime * 1000;
```

Comment:
I am ok with this "as-is". However if it helps to improve quality of calculation (and be more natural) we can store `_historicalPartialGCTime` in microseconds and use /1000 where it necessary. This time comes from PGC time converted to milliseconds in `MM_SchedulingDelegate::partialGarbageCollectCompleted()` - line touched in this change set earlier

Reply:
Right I agree that In some places it isn't very natural. My original thought was to keep `_historicalPartialGCTime ` in ms (since it is used elsewhere), and convert to us where necessary, but it seems like there are more places where it is used in us now...

I think your idea of doing this in a separate change might be best, that way it can be the main focus on the change, and given proper attention